src/JuMP.jl

```diff
@@ -338,9 +336,8 @@ abstract type AbstractConstraint end
 # In JuMP, used only for Variable. Useful primarily for extensions
 abstract type AbstractJuMPScalar end
 
-Base.start(::AbstractJuMPScalar) = false
-Base.next(x::AbstractJuMPScalar, state) = (x, true)
-Base.done(::AbstractJuMPScalar, state) = state
+Base.iterate(x::AbstractJuMPScalar) = (x, true)
```

Comment:
We should do `if` `else` with Julia version and keep start/next/done on v0.6

Reply:
I didn't know there was a `Compat.iterate`, could you point me to where it's defined ?